To add comments to clarify WCS examples

kakirastern opened the pull request:
To follow up on Issue #3840, add additional comments to Astropy documentation at [https://astropy.readthedocs.io/en/latest/wcs/index.html#loading-wcs-information-from-a-fits-file](https://astropy.readthedocs.io/en/latest/wcs/index.html#loading-wcs-information-from-a-fits-file) on `wcs` to clarify some details regarding pixel coordinates.

saimn commented:
I think we should also replace the `np.float_` by `dtype=float` to use a more common idiom.

pllim commented on docs/wcs/examples/from_file.py:
```diff
@@ -20,8 +20,9 @@ def load_wcs_from_file(filename):
-    # Note we've silently assumed a NAXIS=2 image here
-    pixcrd = np.array([[0, 0], [24, 38], [45, 98]], np.float_)
+    # Note we've silently assumed a NAXIS=2 image here.
+    # Note also that the pixel coordinates are pairs of [X, Y].
```
@nden, it is best that you have a look at the wordings to make sure they are accurate here. 🙏 
```suggestion
    # Note also that the pixel coordinates are pairs of [X, Y]
    # and since WCS built from FITS header automatically has
    # origin set to 1, [0, 0] is actually not inside the image.
```

pllim commented on docs/wcs/examples/programmatic.py:
```diff
@@ -18,7 +18,10 @@
 # Some pixel coordinates of interest.
-pixcrd = np.array([[0, 0], [24, 38], [45, 98]], np.float_)
+# Note also that the pixel coordinates are pairs of [X, Y]
+# FITS WCS usually uses [1, 1] as the origin, as it hails from the Fortran tradition.
```
Just want to be consistent with above...
```suggestion
# and since WCS built from FITS header automatically has
# origin set to 1, [0, 0] is actually not inside the image.
```

pllim commented on docs/wcs/examples/programmatic.py:
```diff
@@ -18,7 +18,10 @@
-pixcrd = np.array([[0, 0], [24, 38], [45, 98]], np.float_)
+# Note also that the pixel coordinates are pairs of [X, Y]
+# FITS WCS usually uses [1, 1] as the origin, as it hails from the Fortran tradition.
+# In Numpy, [0, 0] is the origin. 
```
```suggestion
```

kakirastern commented:
@pllim Am looking into the matter of squashing... will get back to you soon.

nden commented on docs/wcs/examples/from_file.py:
```diff
@@ -20,8 +20,11 @@ def load_wcs_from_file(filename):
+    # Note we've silently assumed a NAXIS=2 image here.
+    # Note also that the pixel coordinates are pairs of [X, Y]
+    # and since WCS built from FITS header automatically has
+    # origin set to 1, [0, 0] is actually not inside the image.
```
The meaning of origin has confused many.
Is it clearer to say something like:
```
Note also that the pixel coordinates are pairs of [X, Y]
and that "origin" is set to the default value of 1 because 
the FITS standard assumes the first pixel in an image has
 coordinates (1,1). This means that the first pixel we pass, (0, 0)
is not in the image. To tell astropy.wcs that these are zero-based coordinates,
we can pass origin=0.

```

pllim replied:
Thanks, @nden!

@kakirastern , can you please apply the suggested verbiage in both places? Thank you for your continued patience!

kakirastern replied:
@pllim Yup, am looking into the matter...

kakirastern replied:
@pllim Think I have committed the changes to the two files concerned. But do you think it would be better to put the explanation somewhere outside the example so it comes after the code as an aside? Meaning I will add the following: 

> Note also that the pixel coordinates are pairs of [X, Y] and that "origin" is set to the default value of 1 because the FITS standard assumes the first pixel in an image has coordinates (1,1). This means that the first pixel we pass, (0, 0) is not in the image. To tell astropy.wcs that these are zero-based coordinates, we can pass origin=0.

I think this bit of information is of some importance to be treated with care. Not everyone is aware of it.

kakirastern commented:
Hi @pllim, I have just rebased and squashed... Hope that everything went well though. If possible please check for me and let me know if my attempt has been successful. Next, I should deal with the comments to be added to the documentation provided by @nden. After some minor editing, I will use the following as the final version if you approve:
> Note also that the pixel coordinates are pairs of [X, Y] and that the "origin" is set to the default value of 1. This is because the FITS standard assumes that the first pixel in an image has the coordinates (1,1), in the Fortran tradition. This means that the first pixel we pass, (0, 0) is not in the image, as Python is a C-like language and uses 0 as the first index by default. To tell astropy.wcs that these are zero-based coordinates, we can pass or set the parameter origin=0.
